### PR TITLE
WIP: added tons of arithmetic operators to T

### PR DESCRIPTION
**Pull request by kurtbrose**

### CODE REVIEW

File: glom/core.py [hunk 1]
```diff
@@ -1364,6 +1403,34 @@ def _t_eval(target, _t, scope):
...
+        else:  # arithmetic operators
+            try:
+                if op == '+':
+                    cur = cur + arg
+                elif op == '-':
+                    cur = cur - arg
+                elif op == '*':
+                    cur = cur * arg
+                #elif op == '#':
+                #    cur = cur // arg  # TODO: python 2 friendly approach?
```
Comment (mahmoud) [inline glom/core.py:1415]: do we need floor division?

Comment (mahmoud) [reply glom/core.py:1415]: Or wait, if both the user and we imported future division, wouldn't this work fine?

Comment (kurtbrose) [reply glom/core.py:1415]: oh doh forgot about that

**Comment by kurtbrose** [comment]
so... thinking about it more, should there be a separate `T_Expr` type?

with this approach you can do

```python
(T.a + 1).a.b
```

I guess there are some cases where this is valid

```python
(T + [1]).__len__()
```

**Comment by kurtbrose** [comment]
another interesting thing -- there are some built-ins whose behavior can be overridden

for example, by implementing `__reversed__()`, `reverse(T)` would work -- no need for `Invoke` or `(T.a, reverse)`

**Comment by mahmoud** [comment]
General comment: I'm in favor of this. We'd talked about opening up all dunders before and kinda hesitated because of the implications. But this adds an explicit coherent set. I like that it doesn't overlap with M (except for the very niche invert case) and in the typical cases I can think of the return type won't change from whatever T itself is.

**Comment by kurtbrose** [comment]
just in case this gets lost in the shuffle for a little while -- whenever this gets picked up again, one of the challenges is to correctly parenthesis expressions in the repr

**Pull request merged**